tests/libvmi/cgroup.go

```diff
@@ -0,0 +1,221 @@
+package libvmi
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"kubevirt.io/client-go/log"
+
+	k6tv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/tests/testsuite"
+
+	hw_utils "kubevirt.io/kubevirt/pkg/util/hardware"
+
+	"kubevirt.io/kubevirt/pkg/virt-controller/services"
+	"kubevirt.io/kubevirt/tests/libnode"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/virt-handler/cgroup"
+	"kubevirt.io/kubevirt/tests/exec"
+)
+
+func GetVmiCgroupVersion(vmi *k6tv1.VirtualMachineInstance) (cgroup.CgroupVersion, error) {
+	pod, err := GetPodByVirtualMachineInstance(vmi, testsuite.GetTestNamespace(vmi))
+	if err != nil {
+		return "", err
+	}
+
+	return GetPodCgroupVersion(pod)
+}
+
+func GetPodCgroupVersion(pod *k8sv1.Pod) (cgroup.CgroupVersion, error) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		return "", err
+	}
+
+	containerName := ""
+	for _, container := range pod.Spec.Containers {
+		if container.Name == "compute" {
+			containerName = container.Name
+			break
+		}
+	}
+
+	if containerName == "" {
+		containerName = pod.Spec.Containers[0].Name
+	}
+
+	const cgroupV2OnlyFile = "/sys/fs/cgroup/cgroup.controllers"
+
+	command := []string{"bash", "-c", "cat " + cgroupV2OnlyFile}
+
+	_, err = exec.ExecuteCommandOnPod(virtClient, pod, containerName, command)
+	if err == nil {
+		return cgroup.V2, nil
+	}
+
+	if strings.Contains(err.Error(), "No such file or directory") {
+		return cgroup.V1, nil
+	}
+
+	return "", fmt.Errorf("unknown error occured: %v", err)
+}
+
+// subsystem is ignored if empty
+func SearchCgroupByName(nodeName, basePath, subsystem, cgroupName string) (cgroupPath string, err error) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		return "", err
+	}
+
+	if subsystem != "" {
+		basePath = filepath.Join(basePath, subsystem)
+	}
+
+	findCmd := []string{"bash", "-c", fmt.Sprintf("find %s -regex .*%s.* -type d", basePath, cgroupName)}
+
+	cgroupPath, stderr, err := libnode.ExecuteCommandOnNodeThroughVirtHandler(virtClient, nodeName, findCmd)
+	if err != nil || stderr != "" {
+		err = fmt.Errorf("cannot find dedicated cgroup path. err: %v. stderr: %s", err, stderr)
+		return
+	}
+
+	cgroupPath = strings.TrimSpace(cgroupPath)
+
+	if cgroupPath == "" {
+		err = fmt.Errorf("couldn't find %s directory", services.DedicatedCpusContainerName)
+		return
+	}
+
+	return
+}
+
+func GetCgroupByContainer(pod *k8sv1.Pod, containerName, subsystem string) (cgroupPath string, err error) {
+	var containerStatus k8sv1.ContainerStatus
+	for _, c := range pod.Status.ContainerStatuses {
+		if c.Name == containerName {
+			containerStatus = c
+		}
+	}
+
+	if containerStatus.Name == "" {
+		err = fmt.Errorf("couldn't find container %s for pod %s", containerName, pod.Name)
+		return
+	}
+
+	cgroupDirName := "crio-" + strings.TrimPrefix(containerStatus.ContainerID, "cri-o://")
```

Comment:
I think the tests should be agnostic to the underlying CRI thus making it possible to run the suite on any k8s distro.

Reply:
Agreed. Mentioned that in the [comment above](https://github.com/kubevirt/kubevirt/pull/8869#issuecomment-1367866278) (point 3).

I guess I have a similar question to above - should we support CRIO/containerd and a list of fixed CRIs then?
potentially, any CRI can name the cgroups however it chooses to, so I'm not sure how this could become truly generic

Reply:
p.s. do you know if it's a convention for the container ID to look like `<CRI>://<ID>`?
If so, it could be handled pretty easily